lock-api/src/main/java/com/palantir/lock/client/TimeLockClient.java

```diff
@@ -76,12 +77,19 @@ public TimestampRange getFreshTimestamps(int numTimestampsRequested) {
     }
 
     @Override
-    public LockImmutableTimestampResponse lockImmutableTimestamp(LockImmutableTimestampRequest request) {
+    public LockImmutableTimestampResponse lockImmutableTimestamp(IdentifiedTimeLockRequest request) {
```

Comment:
Is this not an API break?

Reply:
The serialized form of `LockImmutableTimestampRequest` and `IdentifiedTimeLockRequest` are the same. We don't use the type name (de)serializer, so the serialized form of *both* of these objects is along the lines of

```
{"requestId":"ed4b106f-8718-4858-a402-d99415124cfe"}
```

(I've also tested this with an old timelock to verify things are compatible.)